### PR TITLE
Added Reset Configuration Feature

### PR DESCRIPTION
**Pull request by tharun208**
Fixes issue #149

@build-canaries/owners

### CODE REVIEW

File: src/client/backup/reset/reset.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    const messages = [ 'Reset your Nevergreen configuration back to defaults. Please note, resetting your configuration can not be undone!' ]
+    const resetConfiguration = () => {
+        window.localStorage.clear()
+        window.location.reload()
+    }
+    return(
+        <div>
+        <Messages type={messageType}
+                messages={messages}
+                data-locator='messages'/>
```
Comment (GentlemanHal) [inline src/client/backup/reset/reset.tsx]: This shouldn't be an error message, a simple `<p>` should do.

File: src/client/backup/reset/reset.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  vertical-align: top; // remove whitespace from the bottom
+  white-space: pre;
+  width: 100%;
+
+  &:not(:disabled) {
+    &:read-write {
+      @include interactive($focus-color-change-percentage: 0%);
+    }
+  }
+}
```
Comment (GentlemanHal) [inline src/client/backup/reset/reset.scss]: These aren't used as far as I can tell

File: src/client/backup/reset/reset.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+.import {
```
Comment (GentlemanHal) [inline src/client/backup/reset/reset.scss]: This name isn't correct, should be related to the component it styles

File: src/client/backup/reset/reset.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import {Messages, MessagesType} from '../../common/Messages'
+export function Reset() {
+    const messageType = MessagesType.ERROR
+    const messages = [ 'Reset your Nevergreen configuration back to defaults. Please note, resetting your configuration can not be undone!' ]
+    const resetConfiguration = () => {
+        window.localStorage.clear()
+        window.location.reload()
+    }
+    return(
+        <div>
```
Comment (GentlemanHal) [inline src/client/backup/reset/reset.tsx]: This should be wrapped in an actual [`Container`](https://github.com/build-canaries/nevergreen/blob/master/src/client/common/Container.tsx)

**Comment by tharun208** [comment]
sure @GentlemanHal

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by tharun208** [comment]
sure @GentlemanHal

File: src/client/backup/reset/reset.tsx [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
+        clear()
         window.location.reload()
     }
     return(
-        <div>
-        <Messages type={messageType}
-                messages={messages}
-                data-locator='messages'/>
-        <PrimaryButton className={styles.import}
+        <>
```
Comment (GentlemanHal) [inline src/client/backup/reset/reset.tsx]: You don't need the Fragment `<>` here, the `Container` can be the top level element

**Comment by tharun208** [comment]
@GentlemanHal, To do make it simple can I spy on the resetConfiguration method. So, It can cover both of the functions inside it

**Comment by GentlemanHal** [comment]
> @GentlemanHal, To do make it simple can I spy on the resetConfiguration method. So, It can cover both of the functions inside it

No because that is the function under test! Spying on that means someone could delete the contents of the function but the test would still pass 😱

**Comment by tharun208** [comment]
Hi @GentlemanHal, I have made the changes and updated the test to spy on both the clear and reload functions

**Comment by GentlemanHal** [comment]
I've rebased the changes back onto master, thanks @tharun208!

The styling was broken so I made a few changes.

**Pull request closed without merging**